source/Patches/EndGameSummary.cs

```diff
@@ -0,0 +1,218 @@
+
+        public static void UpdatePlayerDataRPCCall()
+        {
+            var writer = AmongUsClient.Instance.StartRpcImmediately(PlayerControl.LocalPlayer.NetId,
+        (byte)CustomRPC.UpdateGamePlayerControlData, Hazel.SendOption.Reliable, -1);
+            AmongUsClient.Instance.FinishRpcImmediately(writer);
+        }
+
+        public static void GatherPlayerData(byte rpccallid)
+        {
+            EndGameSummary.allPlayers.Clear();
+            EndGameSummary.allPlayers = EndGameSummary.GetAllPlayers();
+            List<PlayerControl> allplayersList = EndGameSummary.allPlayers;
+            UpdatePlayerInfo(allplayersList);
+        }
+
+        public static void UpdatePlayerInfo(List<PlayerControl> players)
+        {
+            List<AdditionalTempData.PlayerInfo> pinfolist = new List<AdditionalTempData.PlayerInfo>();
+            pinfolist.Clear();
+
+            foreach (PlayerControl player in players)
+            {
+                int pid = player.PlayerId;
+                string name = player.name;
+                Role role = Role.GetRole(player);
+                Modifier modifier = Modifier.GetModifier(player);
+                Faction faction = role.Faction;
+                int tasksdone = 0;
+                int taskstotal = 0;
+
+                AdditionalTempData.PlayerInfo pinfo = new AdditionalTempData.PlayerInfo();
+                pinfo.PlayerName = name;
+                pinfo.CurrentRole = role;
+                pinfo.Faction = faction;
+                pinfo.Modifier = modifier;
+
+                PluginSingleton<TownOfUs>.Instance.Log.LogMessage($"PlayerName:- " + player.name);
+                foreach (var task in player.myTasks)
+                {
+                    var firstText = task.name;//.Cast<ImportantTextTask>();
+                    PluginSingleton<TownOfUs>.Instance.Log.LogMessage($"TaskName:- " + task.name);
+                    var modname = "junktonull";
+                    var rolename = "junktonull";
+                    if (!IsNull(pinfo, "Mod"))
+                        modname = modifier.Name;
+
+                    if (!IsNull(pinfo, "Role"))
+                        rolename = role.Name;
+
+                    if (!firstText.Contains(modifier.Name) && !firstText.Contains(role.Name) && !firstText.Contains("_Player"))
+                    {
+                        if (task.IsComplete == true)
+                        {
+                            tasksdone++;
+                        }
+                        taskstotal++;
+                    }
+                }
+
+                PluginSingleton<TownOfUs>.Instance.Log.LogMessage($"TaskDone:- " + tasksdone);
+                PluginSingleton<TownOfUs>.Instance.Log.LogMessage($"TaskTotal:- " + taskstotal);
+                pinfo.TasksCompleted = tasksdone;
+                pinfo.TasksTotal = taskstotal;
+                pinfolist.Add(pinfo);
+                tasksdone = 0;
+                taskstotal = 0;
```

Comment:
Why is this reassigned to 0 and never used again?

Reply:
coding i put in because i was getting treble task counts, but i realised i did something wrong and forgot to remove them. thanks for pointing this out.